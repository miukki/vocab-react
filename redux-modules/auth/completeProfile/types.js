const prefix = 'auth/completeProfile'

const types = {}
types.set = function(type) {
  this[type] = `${prefix}/${type}`
}

types.set('SHOW')
types.set('HIDE')

types.set('CHANGE_NAME')
types.set('SUBMIT')
types.set('SUBMIT_DONE')
types.set('SHOW_ERROR')
types.set('DISMISS_ERROR')
types.set('RESET_STATE')

export default types
