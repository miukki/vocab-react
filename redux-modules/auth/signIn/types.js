const prefix = 'auth/signIn'

const types = {}
types.set = function(type) {
  this[type] = `${prefix}/${type}`
}

types.set('SHOW')
types.set('HIDE')

types.set('CHANGE_MOBILE')
types.set('CHANGE_PASSWORD')
types.set('SHOW_PASSWORD')
types.set('HIDE_PASSWORD')
types.set('SUBMIT')
types.set('SUBMIT_DONE')
types.set('SHOW_ERROR')
types.set('DISMISS_ERROR')
types.set('GO_FORGOT_PASSWORD')
types.set('RESET_STATE')

export default types
