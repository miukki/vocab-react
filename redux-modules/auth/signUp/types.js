const prefix = 'auth/signUp'

const types = {}
types.set = function(type) {
  this[type] = `${prefix}/${type}`
}

types.set('SHOW')
types.set('HIDE')

types.set('CHANGE_MOBILE')
types.set('SEND_VCODE')
types.set('SEND_VCODE_DONE')
types.set('COUNTDOWN_VCODE')
types.set('CHANGE_VCODE')
types.set('CHANGE_PASSWORD')
types.set('SHOW_PASSWORD')
types.set('HIDE_PASSWORD')
types.set('SUBMIT')
types.set('SUBMIT_DONE')
types.set('SHOW_ERROR')
types.set('DISMISS_ERROR')
types.set('RESET_STATE')

export default types
