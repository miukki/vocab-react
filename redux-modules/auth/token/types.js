const prefix = 'auth/token'

const types = {}
types.set = function(type) {
  this[type] = `${prefix}/${type}`
}

types.set('SET_JWT')
types.set('DELETE_JWT')

export default types
