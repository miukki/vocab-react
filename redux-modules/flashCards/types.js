export const GENERATE_NEW = '@@reviewItems/GENERATE_NEW'
export const UPDATE_INDEX = '@@reviewItems/UPDATE_INDEX'
export const FLIP = '@@reviewItems/FLIP'
