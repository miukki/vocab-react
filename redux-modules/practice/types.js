export const SELECT_ANSWER = '@@practice/select-answer'
