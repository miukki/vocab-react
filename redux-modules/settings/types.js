export const UPDATE_LOCATION = '@@settings/UPDATE_LOCATION'
