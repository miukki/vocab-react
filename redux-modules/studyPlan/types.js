const prefix = 'eic-vocab-app/studyPlan'

const types = {}
types.set = function(type) {
  this[type] = `${prefix}/${type}`
}

// picker
types.set('UPDATE_PICKER_DAILY_WORDS_AMOUNT')
types.set('UPDATE_PICKER_DAYS_AMOUNT')

// user data
types.set('LOAD_SAVED_DATA')
types.set('LOAD_SAVED_DATA_DONE')

// change user data
types.set('CHANGE_WORDS_LIST')
types.set('UPDATE_WORDS_LIST')

types.set('CONFIRM_PICKER')
types.set('UPDATE_DAILY_WORDS_AMOUNT')

types.set('ADD_LEARNED_WORD')
types.set('UPDATE_LEARNED_WORDS')

types.set('ADD_CHECKIN_LOG')
types.set('UPDATE_CHECKIN_LOGS')

types.set('UPDATE_USER_DATA') // for reducer update user data state

// save user data
types.set('SAVE_USER_DATA_DONE')

// others
types.set('UPDATE_CURRENT_DATE')
types.set('CHECK_DATE')

export default types
