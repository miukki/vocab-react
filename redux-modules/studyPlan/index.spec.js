import { fromJS } from 'immutable'

import reducer, { initialState } from './'
import types from './types'

describe('studyPlan redux', function() {
  it('should return the initial state', function() {
    expect(reducer(undefined, {})).toMatchSnapshot()
  })

  it('should update currentDate', function() {
    Date.prototype.toISOString = jest.fn()
    Date.prototype.toISOString.mockReturnValue('2018-05-01')

    expect(
      reducer(fromJS(initialState), {
        type: types.UPDATE_CURRENT_DATE
      }).toJS()
    ).toMatchSnapshot()
  })
})
