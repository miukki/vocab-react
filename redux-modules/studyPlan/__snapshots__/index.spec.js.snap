// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`studyPlan redux should return the initial state 1`] = `
Immutable.Map {
  wordsListUid: "",
  learnedWords: Set {},
  checkinLogs: Set {},
  dailyWordsAmountOfWordsList: Immutable.Map {
  },
  picker: Immutable.Map {
    dailyWordsAmount: 15,
  },
  currentDate: null,
}
`;

exports[`studyPlan redux should update currentDate 1`] = `
Object {
  "checkinLogs": Set {},
  "currentDate": "2018-05-01",
  "dailyWordsAmountOfWordsList": Object {},
  "learnedWords": Set {},
  "picker": Object {
    "dailyWordsAmount": 15,
  },
  "wordsListUid": "",
}
`;
